llvm/include/llvm/Support/Caching.h

```diff
@@ -54,9 +54,29 @@ using AddStreamFn = std::function<Expected<std::unique_ptr<CachedFileStream>>(
 ///
 /// if (AddStreamFn AddStream = Cache(Task, Key, ModuleName))
 ///   ProduceContent(AddStream);
-using FileCache = std::function<Expected<AddStreamFn>(
+using FileCacheFunction = std::function<Expected<AddStreamFn>(
     unsigned Task, StringRef Key, const Twine &ModuleName)>;
 
+struct FileCache {
```

Comment:
Should we move the docs for `FileCacheFunction` to `FileCache` since I think that will be the primary struct we should use. Also, we should add brief docs for `FileCacheFunction` to explain the arguments.